src/Compilers/VisualBasic/Portable/Binding/Binder_ObjectInitializer.vb

```diff
@@ -508,6 +517,73 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Return resultExpression
         End Function
 
+        Private Shared Sub CheckRequiredMembersInObjectInitializer(
+            constructor As MethodSymbol,
+            containingType As NamedTypeSymbol,
+            initializers As ImmutableArray(Of BoundExpression),
+            creationSyntax As SyntaxNode,
+            diagnostics As BindingDiagnosticBag)
+
+            ' The only time constructor will be null is if we're trying to invoke a parameterless struct ctor, and it's not accessible (such as being protected).
+            Debug.Assert((constructor IsNot Nothing AndAlso ReferenceEquals(constructor.ContainingType, containingType)) OrElse containingType.IsStructureType())
+
+            If constructor IsNot Nothing AndAlso constructor.HasSetsRequiredMembers Then
+                Return
+            End If
+
+            If containingType.HasRequiredMembersError Then
+                ' A use-site diagnostic will be reported on the use, so we don't need to do any more checking here.
+                Return
+            End If
+
+            If TypeOf containingType Is SourceMemberContainerTypeSymbol Then
+                ' A diagnostic will be reported for inheriting from a type with required members, no need to add
+                ' extra diagnostics here.
+                Debug.Assert(Not containingType.HasAnyDeclaredRequiredMembers)
+                Return
+            End If
+
+            Dim requiredMembers = containingType.AllRequiredMembers
+
+            If requiredMembers.Count = 0 Then
+                Return
+            End If
+
+            Dim requiredMembersBuilder = requiredMembers.ToBuilder()
+
+            If Not initializers.IsDefaultOrEmpty Then
+                For Each initializer In initializers
+                    Dim assignmentOperator = TryCast(initializer, BoundAssignmentOperator)
+                    If assignmentOperator Is Nothing Then
+                        Continue For
+                    End If
+
+                    Dim memberSymbol As Symbol = If(
+                        DirectCast(TryCast(assignmentOperator.Left, BoundPropertyAccess)?.PropertySymbol, Symbol),
+                        TryCast(assignmentOperator.Left, BoundFieldAccess)?.FieldSymbol)
+
+                    If memberSymbol Is Nothing Then
+                        Continue For
+                    End If
+
+                    Dim requiredMember As Symbol = Nothing
+                    If Not requiredMembersBuilder.TryGetValue(memberSymbol.Name, requiredMember) Then
+                        Continue For
+                    End If
+
+                    If Not memberSymbol.Equals(requiredMember, TypeCompareKind.AllIgnoreOptionsForVB) Then
```

Comment:
>If Not memberSymbol.Equals(requiredMember, TypeCompareKind.AllIgnoreOptionsForVB) Then [](http://example.com/codeflow?start=20&length=86)

Are we testing scenarios when ```memberSymbol``` is an override?

Reply:
`EnforcedRequiredMembers_Inheritance_*`